test/offchainAssetVault/OffChainAssetVault.test.ts

```diff
@@ -150,4 +170,581 @@ describe("OffChainAssetVault", async function () {
+      `wrong caller expected ${alice.address} got ${caller}`
+    );
+    assert(
+      confiscatee === alice.address,
+      `wrong confiscatee expected ${alice.address} got ${confiscatee}`
+    );
+  });
+  it("Confiscate overloaded - Checks ConfiscateShares is emitted", async function () {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+    const [vault] = await deployOffChainAssetVault();
+    const [receiptVault, asset, priceOracle] =
+      await deployERC20PriceOracleVault();
+
+    const _id = await priceOracle.price();
+
+    await vault.grantRole(await vault.CONFISCATOR(), alice.address);
+
+    const { caller, confiscatee, id } = (await getEventArgs(
+      await vault["confiscate(address,uint256)"](alice.address, _id),
+      "ConfiscateReceipt",
+      vault
+    )) as ConfiscateReceiptEvent["args"];
+
+    assert(
+      caller === alice.address,
+      `wrong caller expected ${alice.address} got ${caller}`
+    );
+    assert(
+      confiscatee === alice.address,
+      `wrong confiscatee expected ${alice.address} got ${confiscatee}`
+    );
+    assert(id.eq(_id), `wrong id expected ${_id} got ${id}`);
+  });
+  it("Checks confiscated is same as balance", async function () {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+    const bob = signers[1];
+    const [vault] = await deployOffChainAssetVault();
+    const [receiptVault, asset, priceOracle] =
+      await deployERC20PriceOracleVault();
+
+    await vault.grantRole(await vault.CONFISCATOR(), alice.address);
+    await vault.grantRole(await vault.CONFISCATOR(), bob.address);
+    await vault.grantRole(await vault.DEPOSITOR(), alice.address);
+
+    const assets = ethers.BigNumber.from(100);
+
+    await asset.transfer(alice.address, assets);
+
+    await asset.connect(alice).increaseAllowance(vault.address, assets);
+
+    const shareRatio = await priceOracle.price();
+    await vault
+      .connect(alice)
+      ["deposit(uint256,address,uint256,bytes)"](
+        assets,
+        bob.address,
+        shareRatio,
+        []
+      );
+
+    const shares = await vault["balanceOf(address)"](bob.address);
+
+    const { confiscated } = (await getEventArgs(
+      await vault.connect(alice)["confiscate(address)"](bob.address),
+      "ConfiscateShares",
+      vault
+    )) as ConfiscateSharesEvent["args"];
+
+    assert(
+      confiscated.eq(shares),
+      `wrong confiscated expected ${shares} got ${confiscated}`
+    );
+  });
+  it("Checks confiscated is transferred", async function () {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+    const bob = signers[1];
+    const [vault] = await deployOffChainAssetVault();
+    const [receiptVault, asset, priceOracle] =
+      await deployERC20PriceOracleVault();
+
+    await vault.grantRole(await vault.CONFISCATOR(), alice.address);
+    await vault.grantRole(await vault.DEPOSITOR(), alice.address);
```

Comment:
probably better to have 3 addresses to be realistic, confiscating from the depositor is weird